Flow.Launcher/MainWindow.xaml.cs

```diff
@@ -267,14 +307,18 @@ private void OnPreviewDragOver(object sender, DragEventArgs e)
             e.Handled = true;
         }
 
-        private void OnContextMenusForSettingsClick(object sender, RoutedEventArgs e)
+        private async void OnContextMenusForSettingsClick(object sender, RoutedEventArgs e)
         {
+            _viewModel.Hide();
+            await Task.Delay(100);
             App.API.OpenSettingDialog();
         }
 
 
-        private void OnDeactivated(object sender, EventArgs e)
+        private async void OnDeactivated(object sender, EventArgs e)
         {
+            _viewModel.Save();
+            await Task.Delay(100);
```

Comment:
> When dragging or closing the setting window, the main window's inactive immediately worked. I added delay in deactive code. because I needed time to initialize the animation.

You are referring to this delay right?

Reply:
```suggestion
            // need time to initialize the main query window animation
            if (_settings.UseAnimation)
                await Task.Delay(100);
```

Reply:
we can add a condition since this is only a requirement for animation.